### PR TITLE
[CCAP-452] - create new providerresponse paid by ccap screen

### PR DESCRIPTION
**Pull request by enyia21**
#### 🔗 Jira ticket
CCAP-452

#### ✍️ Description
Add paid-for-ccap screen and tests with element id's for each button and/or link on the page.  

#### 📷 Design reference
[Figma](https://www.figma.com/design/lGFsTvWwHoOOJ9XQGPFNmu/IL-CCAP-Provider-Experiences-(WIP)?node-id=1448-20273&t=GHbWdMur4Bhsh4Gb-4)

#### ✅ Completion tasks

- [ ] Added relevant tests
- [ ] Meets acceptance criteria

### CODE REVIEW

File: src/main/resources/templates/providerresponse/paid-by-ccap.html [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
       <main id="content" role="main" class="form-card spacing-above-35">
-        <th:block th:replace="~{fragments/cardHeader :: cardHeader(header=#{placeholder}, subtext=#{placeholder})}"/>
+        <th:block th:replace="~{fragments/gcc-icons :: caring}"></th:block>
+        <th:block th:replace="~{fragments/cardHeader :: cardHeader(header=#{paid-by-ccap.header})}"/>
         <th:block th:replace="~{fragments/form :: form(action=${formAction}, content=~{::formContent})}">
           <th:block th:ref="formContent">
             <div class="form-card__content">
-                <!-- Put inputs here -->
+                <th:block th:replace="~{fragments/inputs/yesOrNo :: yesOrNo(
+                  inputName='providerPaidCcap',
```
[the remark this answers is not in the document]
Comment (enyia21) [reply src/main/resources/templates/providerresponse/paid-by-ccap.html:17]: Where should it go?

[the remark this answers is not in the document]
Comment (cram-cfa) [reply src/main/resources/templates/providerresponse/paid-by-ccap.html:17]: @enyia21 @analoo it should go to the 'registration-start' screen with "No" or "I'm not sure" are clicked, right? Am I reading the original comment correctly?

[the remark this answers is not in the document]
Comment (analoo) [reply src/main/resources/templates/providerresponse/paid-by-ccap.html:17]: Yes that's right @cram-cfa

File: src/main/java/org/ilgcc/app/submission/conditions/IsPaidByCCAP.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import formflow.library.config.submission.Condition;
+import formflow.library.data.Submission;
+import org.springframework.stereotype.Component;
+
+@Component
+public class IsPaidByCCAP implements Condition {
+
+    @Override
+    public Boolean run(Submission submission) {
+        return submission != null && submission.getInputData() != null && "true".equals(
```
Comment (analoo) [inline src/main/java/org/ilgcc/app/submission/conditions/IsPaidByCCAP.java:12]: submission should never be null.  and submission.getInputData() should never be null.

We can check:


submission.getInputData().containsKey("providerPaidCcap") && "true".equals(submission.getInputData().get("providerPaidCCAP"))

Comment (cram-cfa) [reply src/main/java/org/ilgcc/app/submission/conditions/IsPaidByCCAP.java]: We don't even need to check the `containsKey` -- the `get` handles if that's null for us, and by putting the `"true"` first in the equals vs:

```submission.getInputData().get("providerPaidCCAP").equals("true")```

we don't have to worry about a NullPointerException

```"true".equals(null)``` is good

```null.equals("true")``` makes a NullPointerException

PR is updated!

**Review by analoo — APPROVED** [review]

**Comment by analoo** [comment]
Tested it and it looks great!

**Pull request merged**